orderer/consensus/etcdraft/chain_test.go

```diff
@@ -2375,21 +2375,26 @@ var _ = Describe("Chain", func() {
 					close(c1.stopped)
 
 					var newLeader, remainingFollower *chain
+					var c2state raft.SoftState
+					var c3state raft.SoftState
 					for newLeader == nil || remainingFollower == nil {
-						var state raft.SoftState
 						select {
-						case state = <-c2.observe:
-						case state = <-c3.observe:
+						case c2state = <-c2.observe:
+						case c3state = <-c3.observe:
 						case <-time.After(LongEventualTimeout):
 							Fail("Expected a new leader to present")
 						}
 
-						if state.RaftState == raft.StateLeader && state.Lead != raft.None {
-							newLeader = network.chains[state.Lead]
-						}
-
-						if state.RaftState == raft.StateFollower && state.Lead != raft.None {
-							remainingFollower = network.chains[state.Lead]
```

Comment:
I don't understand how it worked all this time. 
How can the follower be the chain instance that is `state.Lead` ?

Reply:
I agree completely... mind boggling